File: src/main/java/me/miki/shindo/injection/interfaces/IMixinMinecraft.java
package me.miki.shindo.injection.interfaces;

import net.minecraft.client.resources.DefaultResourcePack;
import net.minecraft.client.resources.ResourcePackRepository;
import net.minecraft.entity.Entity;
import net.minecraft.util.Session;
import net.minecraft.util.Timer;

import java.io.File;

public interface IMixinMinecraft {
    boolean isRunning();

    Timer getTimer();

    void setSession(Session session);

    void callClickMouse();

    void callRightClickMouse();

    DefaultResourcePack getMcDefaultResourcePack();

    void resizeWindow(int width, int height);

    Entity getRenderViewEntity();

    File getFileResourcepacks();

    ResourcePackRepository getMcResourcePackRepository();

    void setMcResourcePackRepository(ResourcePackRepository repo);
}

File: src/main/kotlin/me/miki/shindo/injection/mixin/interfaces/client/IMixinMinecraft.kt
package me.miki.shindo.injection.mixin.interfaces.client

import java.io.File

/**
 * Interface mixin para Minecraft. Usa Any para tipos do Minecraft para evitar
 * LinkageError quando a interface é carregada por um classloader diferente.
 */
interface IMixinMinecraft {
    fun isRunning(): Boolean

    fun getTimer(): Any

    fun setSession(session: Any)

    fun callClickMouse()

    fun callRightClickMouse()

    fun getMcDefaultResourcePack(): Any

    fun resizeWindow(width: Int, height: Int)

    fun getRenderViewEntity(): Any

    fun getFileResourcepacks(): File

    fun getMcResourcePackRepository(): Any

    fun setMcResourcePackRepository(repo: Any)
}
